type_name.h

```diff
@@ -0,0 +1,37 @@
+#include <string_view>
+
+/*
+
+Header file code copied from stackoverflow answer to the question
+"Is it possible to print a variable's type in standard C++ ? "
+
+This is for C++17
+
+https://stackoverflow.com/questions/81870/is-it-possible-to-print-a-variables-type-in-standard-c/56766138#56766138
+
+Usage:
+
+type_name<decltype(__variable__name__here__)>()
+
+*/
+
+template <typename T>
+constexpr auto type_name() noexcept {
+  std::string_view name, prefix, suffix;
+#ifdef __clang__
+  name = __PRETTY_FUNCTION__;
+  prefix = "auto type_name() [T = ";
+  suffix = "]";
+#elif defined(__GNUC__)
+  name = __PRETTY_FUNCTION__;
+  prefix = "constexpr auto type_name() [with T = ";
+  suffix = "]";
+#elif defined(_MSC_VER)
+  name = __FUNCSIG__;
+  prefix = "auto __cdecl type_name<";
+  suffix = ">(void) noexcept";
+#endif
+  name.remove_prefix(prefix.size());
+  name.remove_suffix(suffix.size());
+  return name;
+}
```

Comment:
This is very clever!

Reply:
Simplified it as we are running replit with the argument -std=c++17, and  lines 22 through 24 are what assign the name, prefix and suffix. Removed lines 25 through 33 as they refer to macros which are meant for other C++ environments where __GNUC__ (GNU compilers that use the C preprocessor) or _MSC_VER  (MS Visual C++ compiler).